libs/file-storage/src/lib/__tests__/file-controller.spec.ts

```diff
@@ -63,45 +71,60 @@ describe('file controller e2e test', () => {
     await localstack.container.stop();
   });
 
-  it(`should get pre assign url successfully, POST ${baseController}/get-upload-pre-assign-url`, async () => {
-    const originalFileNames = generateFileNames(4);
-
-    const response = await app.inject({
-      method: 'POST',
-      url: `${baseController}/get-upload-pre-assign-url`,
-      payload: {
-        originalFileNames,
-      },
-    });
-    expect(response.statusCode).toEqual(HttpStatus.OK);
-
-    const createResponseBody = response.json<PreSignedResponse[]>();
-
-    expect(createResponseBody.length).toBe(4);
-
-    for (const [index, resBody] of createResponseBody.entries()) {
-      expect(resBody.key).toContain(originalFileNames[index]);
-      expect(resBody.preSignedUrl).toContain(originalFileNames[index]);
-      expect(resBody.originalFileName).toBe(originalFileNames[index]);
-    }
-  });
+  it.each([
+    {
+      filesData: generateFileNames(4),
+    },
+    {
+      filesData: generateFileNames(4, faker.string.uuid()),
+    },
+  ])(
+    `should get pre assign url successfully, POST ${baseController}/get-upload-pre-assign-url`,
+    async ({ filesData }) => {
+      const response = await app.inject({
+        method: 'POST',
+        url: `${baseController}/get-upload-pre-assign-url`,
+        payload: {
+          filesData,
+        } as UploadPresignRequest,
+      });
+      expect(response.statusCode).toEqual(HttpStatus.OK);
+
+      const createResponseBody = response.json<PreSignedResponse[]>();
+
+      expect(createResponseBody.length).toBe(4);
+
+      for (const [index, resBody] of createResponseBody.entries()) {
+        expect(resBody.key).toContain(filesData[index].originalFileName);
+        expect(resBody.preSignedUrl).toContain(
+          filesData[index].originalFileName,
+        );
+        expect(resBody.originalFileName).toBe(
+          filesData[index].originalFileName,
+        );
+        expect(
+          resBody.key.startsWith(filesData[index].folder || ''),
+        ).toBeTruthy();
+      }
+    },
+  );
 
   it.each([
     {
-      originalFileNames: [],
+      filesData: [],
     },
     {
-      originalFileNames: generateFileNames(22),
+      filesData: generateFileNames(22),
     },
   ])(
     `should not get pre assign url, because of the error. POST ${baseController}/get-upload-pre-assign-url`,
```

Comment:
should be GET in name I guess

Reply:
and in name you can be more specific, like could not get presign url 

also please check names everywhere, it's not pre assign it's presign